lib/evmone/eof.hpp

```diff
@@ -65,6 +65,14 @@ struct EOF1Header
         return container.substr(data_offset);
     }
 
+    /// A helper to check whether the container can be an initcontainer.
+    [[nodiscard]] bool can_init(bytes_view container) const noexcept
+    {
+        // Containers with truncated data section cannot be initcontainers.
```

Comment:
I would add at least
```cpp
assert(static_cast<size_t>(data_offset + data_size) >= container.size());
```
or just made the check to use `!=`

Reply:
yea, I had a 2nd thought to do this with `!=` but didn't get around to fixing this with all the rebase stuff now